### PR TITLE
Feature activemq

### PR DESCRIPTION
**Pull request by luckyss6**
#29

### CODE REVIEW

**Comment by luckyss6** [comment]
@bugrakocabay 
image
i can't merge into dev branch At least 1 approving review is required by reviewers with write access.

**Comment by luckyss6** [comment]
and the problem is this library. https://github.com/go-stomp/stomp
I found a post about this question. https://stackoverflow.com/questions/39914161/how-to-solve-go-stomp-read-timeout
I don't think it's good to be controlled by the HeartBeat option. If readtimeout err, should we join the retry mechanism?

File: pkg/config/provider.go [hunk 1]
```diff
@@ -144,3 +178,22 @@ func (k *KafkaConfig) validateKafkaConfig() error {
...
+	if s.Port == 0 {
+		return stompPortNotDefinedError
+	}
+
+	if len(s.Username) == 0 {
+		return stompUsernameNotDefinedError
+	}
+
+	if len(s.Password) == 0 {
+		return stompHostNotDefinedError
```
Comment (bugrakocabay) [inline pkg/config/provider.go]: Here, `stompPasswordNotDefinedError` should be used.

File: pkg/queue/activemq/activemq.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		stomp.ConnOpt.HeartBeat(2*time.Hour, 2*time.Hour),
+		stomp.ConnOpt.HeartBeatError(5 * time.Minute),
+		stomp.ConnOpt.Login(c.config.Username, c.config.Password),
+	}
+	var connectionString = net.JoinHostPort(c.config.Host, strconv.Itoa(c.config.Port))
+
+	c.conn, err = stomp.Dial("tcp", connectionString, options...)
+	if err != nil {
+		return err
+	}
```
Comment (bugrakocabay) [inline pkg/queue/activemq/activemq.go:39]: Can you add a log message here, something like;
`slog.Info("Connected to ActiveMQ", "host", c.config.Host, "port", c.config.Port)`

File: pkg/queue/activemq/activemq.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	return &Consumer{
+		config: cfg,
+	}
+}
+
+func (c *Consumer) Connect(ctx context.Context) error {
+	slog.Debug("Attempting to connect to ActiveMQ", "host", c.config.Host, "port", c.config.Port)
+	var err error
+	var options = []func(*stomp.Conn) error{
+		//set rw timeout to 2 hours
```
Comment (bugrakocabay) [inline pkg/queue/activemq/activemq.go]: Let's remove this comment.

**Review by bugrakocabay — CHANGES_REQUESTED** [review]
Hi @luckyss6 ,

Thank you for your effort, overall it looks great and working fine. There are some small changes required. And also can you add test cases to `config_test.go` that covers your changes on `provider.go`.

File: pkg/config/provider.go (hunk 1 above)
Comment (luckyss6) [reply pkg/config/provider.go]: I resolved it

File: pkg/queue/activemq/activemq.go (hunk 2 above)
Comment (luckyss6) [reply pkg/queue/activemq/activemq.go:39]: I resolved it

File: pkg/queue/activemq/activemq.go (hunk 3 above)
Comment (luckyss6) [reply pkg/queue/activemq/activemq.go]: I resolved it

**Comment by luckyss6** [comment]
@bugrakocabay i add config_test

**Review by bugrakocabay — APPROVED** [review]
Lgtm! Thank you @luckyss6 for your contribution to konsume.

**Pull request merged**